Design/SANS/NewISISDesign/Design_Document.md

```diff
@@ -0,0 +1,350 @@
+
+![](./Images/SANSConvertToQISIS.png)
+
+## Workspace loading
+
+The logic for loading workspaces has been a sizable part of the original SANS reduction code base. The main reason for this is the fact that there are many different data input formats such as:
+* *.raw* files
+* *.nxs* files with histogram-type information
+* *.nxs* files with event-type information
+* multiperiod data for each of the types above
+* added histogram-type data
+* added event-type data
+
+which can be used for
+* the data workspace
+* the data monitor workspace
+* the transmission/direct workspace
+
+Unlike many other applications in Mantid, loading has gone traditionally through the ADS. Here, this feature is being kept, since it is desirable to avoid reloading data (see [R.3.1](#User_Requirements)). This optimization should be applicable to all data. Nevertheless we should be able to run the entire reduction without any dependence on the ADS. Hence we use it as an option.
+
+In order to reuse the data which is being requested by a reduction, the new work-flow will look for untouched copies of a workspace (i.e. they should not have been moved). It is impossible to judge if a loaded workspace has been altered by the user. The only thing that can be done is to reset the workspace to its original position, if move operations have been applied to it.
+
+In the most common case, we should not have to worry about altered workspaces, since the reduction does not touch the workspaces themselves. There will be however an option on the loaders to allow users to apply a move operation on the workspaces while loading. This is required since
+users want to be able to check the detector position on the fly (see [R.3.5.](#User_Requirements)). It is essentially used as a diagnostic tool.
+
+A general algorithm `SANSLoadData` will receive a `SANSState` object from which it will decide via a `SANSLoaderFactory` which type of `SANSLoader` to  instantiate. This will allow for facility-dependent adjustments should they be required. A `MoveWorkspace` option will allow for moving the workspaces on initial loading should the user explicitly want to do so.
+
+See the details in the image below.
+
+![](./Images/SANSLoadData.png)
+
+
+The `SANSLoader` will loop over all defined data workspaces and load them with the appropriate command. Note that we need to distinguish between loading transmissions and loading scatter data. The `SANSLoader` is lastly also responsible for searching for a workspace on the ADS and for publishing a workspace to the ADS if this has not been done yet. The two diagrams below show the basic logic flow of the actual loaders.
+
+![](./Images/SANSLoader.png)
+
+![](./Images/SANSLoader_load.png)
+
+
+The individual loaders themselves will determine which loading strategy to apply to the file in question, e.g. multiperiod, *.raw* etc.
+
+## Interfacing the new reducer system with the Python commands
+
```

Comment:
Given the time-scales for implementing this. I hope that @SimonHeybrock's `pythonapi` is the target module used, and I think that this should be designed to use that interface not the `simpleapi`